src/Microsoft.Identity.Client/Platforms/net45/WebUI.cs

```diff
@@ -35,8 +35,18 @@ internal abstract class WebUI : IWebUI
 
             var sendAuthorizeRequestWithTcs = new Action<object>((tcs) =>
             {
-                authorizationResult = Authenticate(authorizationUri, redirectUri);
-                ((TaskCompletionSource<object>)tcs).TrySetResult(null);
+                try
+                {
+                    authorizationResult = Authenticate(authorizationUri, redirectUri);
+                }
+                catch (Exception e)
+                {
+                    // Need to catch the exception here and put on the TCS which is the task we are waiting on so that
+                    // the exception comming out of Authenticate is correctly thrown.
+                    ((TaskCompletionSource<object>)tcs).TrySetException(e);
+                }
+
+                 ((TaskCompletionSource<object>)tcs).TrySetResult(null);
```

Comment:
I am actually going to move this onto line 41

Reply:
Was about to comment on that.